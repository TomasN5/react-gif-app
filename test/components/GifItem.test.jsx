/* eslint-env jest */
import { describe, test, expect } from '@jest/globals';
import { render,screen } from '@testing-library/react';
import GifItem from '../../src/Components/GifItem';

describe('Pruebas en GifItem', () => {
  
  const url = "http://localhost/Hola" 
  const title = "Chau"

  test('Debe hacer match con el snapshot', () => {
    const {asFragment} = render(<GifItem title={title} url={url}/>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Debe mostrar la imagen con el url y el ALT indicado', () => {
    render(<GifItem title={title} url={url}/>)
    //expect(screen.getByRole('img').src).toBe(url);
    const {src,alt}=screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })
  test('Debe mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url}/>)
    expect(screen.getByText(title)).toBeTruthy()
  })
  
  
})
