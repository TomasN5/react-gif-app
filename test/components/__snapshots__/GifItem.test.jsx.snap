// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Chau"
      src="http://localhost/Hola"
    />
    <p>
      Chau
    </p>
  </div>
</DocumentFragment>
`;
